Code/src/main/resources/catalog/catalog.xml

```diff
@@ -9,15 +9,35 @@
             <instance name="Double"/>
             <instance name="Char"/>
             <instance name="Bool"/>
+            <instance name="[]" constrainedArgs="1"/>
```

Comment:
Maybe it would be nicer to have the instance name as "[a]" or "[] a" or "Either a b" here? These instance names go through the type parser anyway, and that way you wouldn't need the constrainedArgs counter.